flent/runners.py

```diff
@@ -845,6 +846,95 @@ def find_binary(cls, host, interval, length, ip_version, local_bind=None,
         raise RuntimeError("No suitable Iperf binary found.")
 
 
+class ParseException(Exception):
+    pass
+
+
+class SsRunner(ProcessRunner):
+    """Runner for iterated `ss -t -i -p`. Depends on same partitial output
+    separationa and time stamping as TcRunner."""
+
+    ip_v4_addr_sub_re = "(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])"
+    ip_v6_addr_sub_re = "(?:(?:[A-F0-9]{1,4}:){7}[A-F0-9]{1,4}|(?=(?:[A-F0-9]{0,4}:){0,7}[A-F0-9]{0,4}(?![:.\w]))(([0-9A-F]{1,4}:){1,7}|:)((:[0-9A-F]{1,4}){1,7}|:))(:\d+)+"
+
+    time_re = re.compile(r"^Time: (?P<timestamp>\d+\.\d+)", re.MULTILINE)
+    cwnd_re = re.compile(r"cwnd:(?P<cwnd>\d+)", re.MULTILINE)
+    rtt_re = re.compile(r"rtt:(?P<rtt>\d+.\d+/\d+.\d+)", re.MULTILINE)
+    pid_re = re.compile(r"pid=(?P<pid>\d+)", re.MULTILINE)
+    ports_ipv4_re = re.compile(r"" + ip_v4_addr_sub_re +
+                               ":(?P<src_p>\d+)\s+?" + ip_v4_addr_sub_re + ":(?P<dst_p>\d+)")
+    ports_ipv6_re = re.compile(r"" + "(?P<src_p>" + ip_v6_addr_sub_re + ")" + "\s+" +
+                               "(?P<dst_p>" + ip_v6_addr_sub_re + ")", re.IGNORECASE)
+
+    src_p = []
+    dst_p = []
+    pid = []
+
+    def parse_part(self, part):
```

Comment:
Right, was still not doing anything according the filtering since of the open API quarrel. Hm, you know your way around better, I know where runners are kicked off, though I'd have to start a grep-marathon still. Thus, I'd appreciate you delineating the framing.